CMakeLists.txt

```diff
@@ -98,9 +98,9 @@ add_dependencies(qa format)
 add_custom_target(build_tests)
 
 # clang-check and clang-format
-# The .clang-format file requires clang-format-3.8
 find_program(CLANG_FORMAT NAMES clang-format-3.8 clang-format)
-find_program(CLANG_CHECK NAMES clang-check-3.8 clang-check)
+find_program(CLANG_CHECK NAMES clang-check-6.0 clang-check-3.8 clang-check)
+find_program(CLANG_TIDY NAMES clang-tidy-6.0 clang-tidy-3.8 clang-tidy)
```

Comment:
Just a word of warning: previously we didn't push past `clang-check-3.8` because we found that between 3.8 and 4 there were some annoying situations where one version forces one thing and the other forces another thing, which meant if you didn't use the same as what the server had, you could almost never get CI to pass. For now, I'm happy to give it a go and see if we have problems again.

Reply:
I see, sound scary...

The recent versions should feature more checks but if they are too painful to install on our ubuntu, we might have to roll them back at some point.